pkg/controller/podgc/gc_controller.go

```diff
@@ -143,13 +146,43 @@ func (gcc *PodGCController) gcTerminated(pods []*v1.Pod) {
 func (gcc *PodGCController) gcOrphaned(pods []*v1.Pod) {
 	klog.V(4).Infof("GC'ing orphaned")
 	// We want to get list of Nodes from the etcd, to make sure that it's as fresh as possible.
-	nodes, err := gcc.kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
-	if err != nil {
+
+	// get nodes from resource provider clients
+	allRpNodes := make(map[string]*v1.NodeList, len(gcc.resourceProviderClients))
```

Comment:
do we have some analysis on how often this function to be be called in our test ( or some other env ) ? if this is called frequently it would be impact GC perf.

Reply:
> do we have some analysis on how often this function to be be called in our test ( or some other env ) ? if this is called frequently it would be impact GC perf.

It's called every 20s.

Reply:
good to know. in this case, then it probably cannot be implemented as such. likely a list/watch approach is needed  here.

Reply:
> good to know. in this case, then it probably cannot be implemented as such. likely a list/watch approach is needed here.

This is a K8s existing implementation. I agree list/watch might have better performance but this code has comment specifically said it wanted real time data. I would rather keep it the original logic before we found there are performance issue.

Reply:
that should be OK for the 430 we can always revisit.